### PR TITLE
Added multimodal embeddings

### PR DESCRIPTION
**Pull request by ivanmkc**
**REQUIRED:** Add a summary of your PR here, typically including why the change is needed and what was changed. Include any design alternatives for discussion purposes.

Added new multimodal embeddings notebook.

**REQUIRED:** Fill out the below checklists or remove if irrelevant
1. If you are opening a PR for `Official Notebooks` under the [notebooks/official](https://github.com/GoogleCloudPlatform/vertex-ai-samples/tree/main/notebooks/official) folder, follow this mandatory checklist:
- [x] Use the [notebook template](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/notebook_template.ipynb) as a starting point.
- [x] Follow the style and grammar rules outlined in the above notebook template.
- [ ] Verify the notebook runs successfully in Colab since the automated tests cannot guarantee this even when it passes.
- [ ] Passes all the required automated checks. You can locally test for formatting and linting with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/CONTRIBUTING.md#code-quality-checks).
- [ ] You have consulted with a tech writer to see if tech writer review is necessary. If so, the notebook has been reviewed by a tech writer, and they have approved it.
- [x] This notebook has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/official/CODEOWNERS) file under the `Official Notebooks` section, pointing to the author or the author's team.
- [x] The Jupyter notebook cleans up any artifacts it has created (datasets, ML models, endpoints, etc) so as not to eat up unnecessary resources.

### CODE REVIEW

**Comment by ivanmkc** [comment]
@sarahcdugan to review text.

**Comment by ivanmkc** [comment]
<p>Not needed. This is already a unique (well, random) bucket name.</p>

---

 <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/2164/discussion/#comment-1673853374'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by ivanmkc** [comment]
<p>Added blurb about batch and generator.</p>

---

 <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/2164/discussion/#comment-1673853458'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Review by andrewferlitsch — APPROVED** [review]

File: notebooks/official/matching_engine/sdk_matching_engine_create_multimodal_embeddings.ipynb [hunk 1]
```diff
@@ -0,0 +1,1533 @@
+{
```
Comment (sarahcdugan) [inline notebooks/official/matching_engine/sdk_matching_engine_create_multimodal_embeddings.ipynb:1]: <p>The last sentence can be changed to something like - "To learn more, see the official documentation for Vertex AI Multimodal Embeddings and Vertex AI Matching Engine."</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/2164/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (sarahcdugan) [inline notebooks/official/matching_engine/sdk_matching_engine_create_multimodal_embeddings.ipynb:1]: <p>"To learn more, see Vertex AI regions."</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/2164/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (sarahcdugan) [inline notebooks/official/matching_engine/sdk_matching_engine_create_multimodal_embeddings.ipynb:1]: <p>I think I'd add the link as a hyperlink. Something like - "..to your service account in the iam gsutil command official documentation" (and link the "iam gsutil command")</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/2164/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

Comment (sarahcdugan) [inline notebooks/official/matching_engine/sdk_matching_engine_create_multimodal_embeddings.ipynb:1]: <p>"Learn more about content detection (SafeSearch) in Detect explicit content" </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/2164/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='38'/>

Comment (sarahcdugan) [inline notebooks/official/matching_engine/sdk_matching_engine_create_multimodal_embeddings.ipynb:1]: <p>Looks like all instances of "safe search" should be changed to "SafeSearch"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/2164/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='40'/>

Comment (sarahcdugan) [inline notebooks/official/matching_engine/sdk_matching_engine_create_multimodal_embeddings.ipynb:1]: <p>"Datasets can be large, so it's recommended to load a batch of data at a time into memory using a generator."</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/2164/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='47'/>

Comment (sarahcdugan) [inline notebooks/official/matching_engine/sdk_matching_engine_create_multimodal_embeddings.ipynb:1]: <p>nice!</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/2164/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='57'/>

Comment (sarahcdugan) [inline notebooks/official/matching_engine/sdk_matching_engine_create_multimodal_embeddings.ipynb:1]: <p>"After you've built"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/2164/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='75'/>

Comment (sarahcdugan) [inline notebooks/official/matching_engine/sdk_matching_engine_create_multimodal_embeddings.ipynb:1]: <p>nit, add a period. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/2164/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='78'/>

**Review by sarahcdugan — APPROVED** [review]

File: notebooks/official/matching_engine/sdk_matching_engine_create_multimodal_embeddings.ipynb (hunk 1 above)
Comment (ivanmkc) [reply notebooks/official/matching_engine/sdk_matching_engine_create_multimodal_embeddings.ipynb:1]: <p>Let's change it in the template if you want this changed. Will keep same to be consistent.</p>

Comment (ivanmkc) [reply notebooks/official/matching_engine/sdk_matching_engine_create_multimodal_embeddings.ipynb:1]: <p>Ditto.</p>

**Pull request merged**